File: Nodal_Discontinuous_Galerkin_Methods_Hesthaven/nudg++/trunk/Src/Codes3D/EquiNodes3D.cpp
// EquiNodes3D.m
// function [X,Y,Z] = EquiNodes3D(N)
// 2007/06/11
//---------------------------------------------------------
#include "NDGLib_headers.h"


//---------------------------------------------------------
void EquiNodes3D(int N, DVec& X, DVec& Y, DVec& Z)
//---------------------------------------------------------
{
  // function [X,Y,Z] = EquiNodes3D(N)
  // Purpose: compute the equidistributed nodes on the 
  //          reference tetrahedron

  // total number of nodes
  int Np = (N+1)*(N+2)*(N+3)/6;  double dN = double(N);

  // create equidistributed nodes on equilateral triangle
  X.resize(Np); Y.resize(Np); Z.resize(Np); 

  int sk = 1;
  for (int n=1; n<=(N+1); ++n) {
    for (int m=1; m<=(N+2-n); ++m) {
      for (int q=1; q<=(N+3-n-m); ++q) {
        X(sk) = -1.0 + (q-1.0)*2.0/dN;
        Y(sk) = -1.0 + (m-1.0)*2.0/dN;
        Z(sk) = -1.0 + (n-1.0)*2.0/dN;
        ++sk;
      }
    }
  }
}

File: Nodal_Discontinuous_Galerkin_Methods_Hesthaven/Codes1.1/Codes3D/EquiNodes3D.m
function [X,Y,Z] = EquinNdes3D(N)

% function [X,Y,Z] = EquinNdes3D(N)
% Purpose: compute the equidistributed nodes on the reference tetrahedron

% total number of nodes
Np = (N+1)*(N+2)*(N+3)/6;

% 2) create equidistributed nodes on equilateral triangle
X = zeros(Np,1); Y = zeros(Np,1); Z = zeros(Np,1); 

sk = 1;
for n=1:N+1
  for m=1:N+2-n
    for q=1:N+3-n-m
      X(sk) = -1 + (q-1)*2/N; Y(sk) = -1 + (m-1)*2/N; Z(sk) = -1 + (n-1)*2/N;
      sk = sk+1;
    end
  end
end
return;
